giturlparse/parser.py

```diff
@@ -82,8 +82,27 @@ def parse(self):
                   '([a-z0-9_.-]*)[:/]*'
                   '([\d]+){0,1}'
                   '([:/](.+)/(.+).git)')
+
         m2 = re.search(regexp, self._url)
 
+        regexp = (r'(git\+)?'
+                  '((?P<protocol>\w+)://)'
+                  '((?P<user>\w+)@)?'
+                  '((?P<site>[\w\.\-]+))'
+                  '(:(?P<port>\d+))?'
+                  '(\/(?P<owner>\w+))?'
+                  '(\/(?P<repo>[\w\-]+)(\.git)?)?')
+
+        m3 = re.search(regexp, self._url)
```

Comment:
should probably solve https://github.com/retr0h/git-url-parse/issues/5 first, so adding more is not more expensive unnecessarily

Reply:
Why are we using named captures?

```
regexp = (r'(git\+)?'
+                  '((?P<protocol>\w+)://)'
+                  '((?P<user>\w+)@)?'
+                  '((?P<site>[\w\.\-]+))'
+                  '(:(?P<port>\d+))?'
+                  '(\/(?P<owner>\w+))?'
+                  '(\/(?P<repo>[\w\-]+)(\.git)?)?')
```

For example <protcol>, <user>, <site>, etc... These named captures are even used.